src/account.ts

```diff
@@ -0,0 +1,843 @@
+   * - Keeping the settlement methods "pure" with respect to the balance of the account greatly
+   *   simplifies the balance logic (but does require committing the balance first, then refunding)
+   */
+  async attemptSettle (): Promise<void> {
+    let amountLeftover = new BigNumber(0)
+    const settleThreshold = this.master._balance.settleThreshold
+    try {
+      // Don't attempt settlement if there's no configured settle threshold ("receive only" mode)
+      if (!settleThreshold) {
+        return this.master._log.trace('Cannot settle: settle threshold must be configured for automated settlement')
+      }
+
+      const shouldSettle = settleThreshold.gt(this.account.balance)
+      if (!shouldSettle) {
+        return this.master._log.trace(`Cannot settle: balance of ${format(this.account.balance, Unit.Gwei)} ` +
+          `is not below settle threshold of ${format(settleThreshold, Unit.Gwei)}`)
+      }
+
+      let settlementBudget = this.master._balance.settleTo.minus(this.account.balance)
+      if (settlementBudget.lte(0)) {
+        // This *should* never happen, since the master constructor verifies that settleTo >= settleThreshold
+        return this.master._log.error(`Critical settlement error: settlement threshold triggered, but settle amount of ` +
+          `${format(settlementBudget, Unit.Gwei)} is 0 or negative`)
+      }
+
+      // If we're not prefunding, the amount should be limited by the total packets we've fulfilled
+      // If we're prefunding, the payoutAmount is infinity, so it doesn't affect the amount to settle
+      settlementBudget = BigNumber.min(settlementBudget, this.account.payoutAmount)
+      if (settlementBudget.lte(0)) {
+        return this.master._log.trace(`Cannot settle: no fulfilled packets have yet to be settled, ` +
+          `payout amount is ${format(this.account.payoutAmount, Unit.Gwei)}`)
+      }
+
+      // This should never error, since settleTo < maximum
+      this.addBalance(settlementBudget)
+      this.account.payoutAmount = this.account.payoutAmount.minus(settlementBudget)
+
+      // Convert gwei to wei
+      settlementBudget = convert(settlementBudget, Unit.Gwei, Unit.Wei).dp(0, BigNumber.ROUND_FLOOR)
+      this.master._log.trace(`Attempting to settle with account ${this.account.accountName} for maximum of ${format(settlementBudget, Unit.Wei)}`)
+
+      try {
+        // 1) Try to send a claim: spend the channel down entirely before funding it
+        amountLeftover = await this.sendClaim(settlementBudget)
+        // 2) Open or deposit to a channel if it's necessary to send anything leftover
+        if (amountLeftover.lte(0)) throw amountLeftover
+        amountLeftover = await this.fundOutgoingChannel(amountLeftover)
+        // 3) Try to send a claim again since we may have more funds in the channel
+        if (amountLeftover.lte(0)) throw amountLeftover
+        amountLeftover = await this.sendClaim(amountLeftover)
+      } catch (leftover) {
```

Comment:
It's strange to throw things that aren't errors. What if `this.sendClaim` actually throws an error? Seems like amountLeftover would be wrong